lib/cocoapods/sources_manager.rb

```diff
@@ -88,11 +88,16 @@ def update(source_name = nil, show_output = false)
         end
 
         changed_spec_paths = {}
-        sources.each do |source|
-          UI.section "Updating spec repo `#{source.name}`" do
-            changed_source_paths = source.update(show_output)
-            changed_spec_paths[source] = changed_source_paths if changed_source_paths.count > 0
-            source.verify_compatibility!
+        # Ceate the Spec_Lock file if needed and lock it so that concurrent
+        # repo updates do not cause each other to fail
+        File.open("#{Config.instance.repos_dir}/Spec_Lock", File::CREAT) do |f|
+          f.flock(File::LOCK_EX)
```

Comment:
Will it lock all repo sources? If I have 10 spec repos, other CI task will wait all repo update finish, right?

Could we only lock the single repo to allow another CI task update other spec repos?

Reply:
I could change it to a per-repo lock file like ‘Spec_Lock-master’ and ‘Spec_Lock-myrepo’. Is that what you’re suggesting?

Reply:
yeah. Why not write the lock file in per-repo directory instead of `Spec_Lock-master` in the root directory?

Reply:
I think we should probably avoid getting those repos dirty

Reply:
we could ignore the file in the .git/exclude/info

Reply:
I found `.git/info/exclude` is that what you mean? If so, what does that do?